src/screens/Movie/Movie.tsx

```diff
@@ -55,13 +60,40 @@ const Movie = ({ match, location }: RouteComponentProps<MovieRouteParams>): JSX.
   const { data: playlist } = usePlaylist(features?.recommendationsPlaylist || '', { related_media_id: id });
 
   // Favorite
-  const isFavorited = useFavoritesStore((state) => !!item && state.hasItem(item));
+  const { isFavorited, favoritesCount } = useFavoritesStore((state) => ({
+    isFavorited: !!item && state.hasItem(item),
+    favoritesCount: state.favorites?.length || 0,
+  }));
 
   // User, entitlement
   const { user, subscription } = useAccountStore(({ user, subscription }) => ({ user, subscription }), shallow);
   const { isEntitled } = useEntitlement(item);
 
   // Handlers
+  const onFavoritesWarningToggle = useCallback(() => {
+    setIsFavoritesWarningShown(!isFavoritesWarningShown);
+  }, [setIsFavoritesWarningShown, isFavoritesWarningShown]);
+
+  const onFavoriteButtonClick = useCallback(() => {
```

Comment:
Would it make sense to put all of this logic into the favoritesController and just expose a `toggleFavorite` method?

Reply:
We should toggle the Alert component state in the callback to show something better than just a native alert message:

<img width="1455" alt="Screenshot 2022-06-14 at 11 19 20" src="https://user-images.githubusercontent.com/27254986/173542644-5c18454d-47ca-47be-b712-729b64791a1c.png">

That is why I decided to use it as a component's callback.

Reply:
I did my best to decouple it and to store it in the controller.